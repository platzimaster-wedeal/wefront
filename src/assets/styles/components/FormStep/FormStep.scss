@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.form-step {
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  overflow: hidden;
  width: 90%;
  border-top: 1px solid $color-black-30;
  border-bottom: 1px solid $color-black-30;
  padding: $space-size-small 0;
  margin: $space-size-medium 0;

  h2 {
    color: $color-black-90;
    transition: 0.2s ease-out;
    line-height: 36px;
    @include font-size($font-size-medium);

    @include media(tablet){
      @include font-size($font-size-medium);
    }
  }

  svg {
    transition: 0.2s ease-out;
  }

  &--complete {
    h2 {
      color: $color-primary;
    }

    svg {
      color: $color-primary;
    }
  }

  &__title {
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: space-between;
    padding: $space-size-medium 0;
    cursor: pointer;

    &:hover {
       h2 {
         color: $color-primary-60;
       }
       svg {
         color: $color-primary-60;
       }
    }
  }

  &__children {
    padding:  0;
    opacity: 1;
    transition: 0.3s ease-in-out;
  }

  &__children--open {
    height: auto;
    padding: $space-size-normal 0;
  }
  &__children--close {
    padding:  0;
    height: 0;
    opacity: 0;
  }

}
