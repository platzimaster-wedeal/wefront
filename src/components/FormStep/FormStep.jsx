import React, { useState } from "react";
import { MdExpandMore, MdExpandLess, MdCheck } from "react-icons/md";

// Styles
import "../../assets/styles/components/FormStep/FormStep.scss";

const FormStep = ({
 title = "Step of form one",
 children,
 isComplete = false,
}) => {
 const [isOpen, setIsOpen] = useState(false);

 const handleOpenState = () => {
  setIsOpen(!isOpen);
 };

 return (
  <div className={`form-step ${isComplete && "form-step--complete"} `}>
   <div className="form-step__title" onClick={handleOpenState}>
    <h2>{title}</h2>

    {isOpen && <MdExpandLess size={30} />}
    {!isOpen && !isComplete && <MdExpandMore size={30} />}
    {!isOpen && isComplete && <MdCheck size={30} />}
   </div>
   <div
    className={`form-step__children form-step__children--${
     isOpen ? "open" : "close"
    }`}>
    {children}
   </div>
  </div>
 );
};

export default FormStep;
