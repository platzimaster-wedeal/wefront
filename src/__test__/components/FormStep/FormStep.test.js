import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import FormStep from "../../../components/FormStep/FormStep";

describe("Test for the component FormStep :D", () => {
 const FormStepMounted = mount(<FormStep />);
 const FormStepCreated = create(<FormStep />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(FormStepCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(FormStepMounted.length).toEqual(1);
  });
  test("Should the description of the User must be a string", () => {
   expect(typeof FormStepMounted.find(".form-step__title").text()).toEqual(
    "string"
   );
  });
 });
});
