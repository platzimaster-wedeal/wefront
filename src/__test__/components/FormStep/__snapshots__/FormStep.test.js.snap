// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component FormStep :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="form-step false "
>
  <div
    className="form-step__title"
    onClick={[Function]}
  >
    <h2>
      Step of form one
    </h2>
    <svg
      fill="currentColor"
      height={30}
      size={30}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={30}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </div>
  <div
    className="form-step__children form-step__children--close"
  />
</div>
`;
